### PR TITLE
remove lib/usr/clib/net directory

### PR DESCRIPTION
**Pull request by KeithWiles**
Remove the clib/net directory/lib and move the headers into lib/include/net
directory. Moving the headers here allows for access to routines for the
commit of TCP. Most of the changes are around include lines changing from
cne_ether.h to net/cne_ether.h or net/cne_arp.h, ...

The lib/includes directory is processed early in building CNDP and the clib/net
directory was being built too late for TCP. Besides the routines were better
off as inline functions, which allows them to be used early.

Removed a couple #if 0 dead code, plus some routines we could convert to standard
inet routines i.e. like _atoip and others.

Also removed a number of duplicate functions, structures and defines.

Signed-off-by: Keith Wiles <keith.wiles@intel.com>

### CODE REVIEW

File: lib/include/net/cne_ether.h [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+ *  the ethernet address.
+ * @param ea2
+ *  A pointer to the second ether_addr structure containing
+ *  the ethernet address.
+ *
+ * @return
+ *  True  (1) if the given two ethernet address are the same;
+ *  False (0) otherwise.
+ */
+static inline int
```
Comment (maryamtahhan) [inline lib/include/net/cne_ether.h:130]: Are the functions in this file intended to be static inline or 
```suggestion
 CNDP_API __cne_always_inline  int
```

File: lib/include/net/cne_ether.h [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+    to_words[2] = from_words[2];
+#else
+    /*
+     * Use the common way, because of a strange gcc warning.
+     */
+    *ea_to = *ea_from;
+#endif
+}
+
+/* eth_swap(uint16_t * to, uint16_t * from) - Swap two 16 bit values */
```
Comment (maryamtahhan) [inline lib/include/net/cne_ether.h:299]:
```suggestion
/* Swap two 16 bit values */
```

File: lib/include/net/cne_ether.h [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+ether_swap(uint16_t *t, uint16_t *f)
+{
+    uint16_t v;
+
+    v  = *t;
+    *t = *f;
+    *f = v;
+}
+
+/* eth_addr_swap( struct ether_addr * to, struct ether_addr * from ) - Swap two
```
Comment (maryamtahhan) [inline lib/include/net/cne_ether.h:310]:
```suggestion
/* Swap two
```

File: lib/include/net/cne_ether.h [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+        /* unknown format */
+    } else
+        return NULL;
+
+    return e;
+}
+
+#ifndef _MTOA_
+#define _MTOA_
+/* char * inet_mtoa(char * buff, int len, struct ether_addr * eaddr) - Convert MAC address to
```
Comment (maryamtahhan) [inline lib/include/net/cne_ether.h:499]:
```suggestion
/* Convert MAC address to
```

File: lib/include/net/cne_net.h [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+#define _CNE_NET_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <net/cne_ip.h>
+#include <net/cne_udp.h>
+#include <net/cne_tcp.h>
+
```
Comment (maryamtahhan) [inline lib/include/net/cne_net.h:15]: Doxygen @file tag missing?

File: lib/include/net/cne_net.h [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+ * mbuf, are not fragmented and can be safely modified.
+ *
+ * @param m
+ *   The packet mbuf to be fixed.
+ * @param ol_flags
+ *   TX offloads flags to use with this packet.
+ * @return
+ *   0 if checksum is initialized properly
+ */
+static inline int
```
Comment (maryamtahhan) [inline lib/include/net/cne_net.h:36]: Just wondering if things here need to also be: 
```suggestion
CNDP_API __cne_always_inline  int
```

**Review by maryamtahhan — APPROVED** [review]

File: lib/cnet/arp/cnet_arp.c [hunk 7]
```diff
@@ -91,12 +92,14 @@ cnet_arp_add(int netif_idx, struct in_addr *addr, struct ether_addr *mac, int pe
...
         if (ret < 0)
-            CNE_WARN("FIB allocate failed for %s\n", inet_ntop4(&entry->pa, &mask));
+            CNE_WARN("FIB allocate failed for %s\n",
+                     inet_ntop4(ipaddr, sizeof(ipaddr), &entry->pa, &mask));
 
         idx = ret;
         if (cne_fib_add(fi->fib, addr->s_addr, 32, idx)) {
             fib_info_free(fi, idx);
-            CNE_NULL_RET("ARP add failed for %s\n", inet_ntop4(&entry->pa, &mask));
+            CNE_NULL_RET("ARP add failed for %s\n",
```
[the remark this answers is not in the document]
Comment (KeithWiles) [reply lib/cnet/arp/cnet_arp.c:101]: Freed the entry pointer.

File: lib/cnet/incs/cnet_rarp.h [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+ * Copyright (c) 2019-2022 Intel Corporation
+ */
+
+#ifndef _CNE_RARP_H_
+#define _CNE_RARP_H_
+
+/**
+ * @file
+ *
+ * ARP-related defines
```
[the remark this answers is not in the document]
Comment (KeithWiles) [reply lib/cnet/incs/cnet_rarp.h:11]: Fixed

File: lib/cnet/netlink/cnet_netlink.c [hunk 9]
```diff
@@ -12,16 +12,16 @@
 #include <arpa/inet.h>
 #include <bsd/string.h>
 
-#include <cne_common.h>        // for
-#include <cne_ether.h>         // for
-#include <cne_thread.h>        // for
+#include <cne_common.h>
```
[the remark this answers is not in the document]
Comment (KeithWiles) [reply lib/cnet/netlink/cnet_netlink.c:15]: Yes, I need to rerun IWYU, but I will do that in another PR.

File: lib/include/net/cne_ether.h (hunk 1 above)
Comment (KeithWiles) [reply lib/include/net/cne_ether.h:130]: Convert them to suggested code.

File: lib/include/net/cne_net.h (hunk 6 above)
Comment (KeithWiles) [reply lib/include/net/cne_net.h:36]: Seems like a number of this function use a static variable, which does not compile changing these to the suggestion above. I will leave this one for now. I was able to change the previous ones to the suggested text.

**Review by maryamtahhan — APPROVED** [review]

File: lib/include/net/cne_ether.h [hunk 10]
```diff
@@ -0,0 +1,507 @@
+/* SPDX-License-Identifier: BSD-3-Clause
```
Comment (jeffreybshaw) [inline lib/include/net/cne_ether.h:1]: This looks like new code, not related to moving header files from one directory to another. Can you put in a different commit? It's hard to complete a review with so much code shuffling aaround.

File: lib/include/net/cne_net.h [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+cne_net_intel_cksum_flags_prepare(pktmbuf_t *m, uint64_t ol_flags)
```
Comment (jeffreybshaw) [inline lib/include/net/cne_net.h:44]: Why does this API have "intel" in the name?

File: lib/include/net/cne_net.h [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+ * This function expects that used headers are in the first data segment of
+ * mbuf, are not fragmented and can be safely modified.
+ *
+ * @param m
+ *   The packet mbuf to be fixed.
+ * @return
+ *   0 if checksum is initialized properly
+ */
+static inline int
+cne_net_intel_cksum_prepare(pktmbuf_t *m)
```
Comment (jeffreybshaw) [inline lib/include/net/cne_net.h:125]: Why is intel in the name?

File: usrtools/txgen/app/cli-functions.c [hunk 13]
```diff
@@ -184,31 +188,26 @@ set_cmd(int argc, char **argv)
...
-                &ip, sizeof(ip));
-            if(ip_ver > 0) {
-                foreach_port(portlist,
-                    single_set_ipaddr(info, 's', &ip, IPv4_VERSION));
-            } else
-                cne_printf("ERROR couldn't set ipv4IP address\n");
+            if (!p)
+                CNE_ERR_RET("src IP address should contain subnet value, default /32 for IPv4, /128 for IPv6\n");
+            *p++ = '\0';
+            prefixlen = atoi(p);
```
Comment (jeffreybshaw) [inline usrtools/txgen/app/cli-functions.c:194]: the result of "atoi" needs to be validated.

File: lib/cnet/incs/cnet_rarp.h [hunk 14]
```diff
@@ -0,0 +1,85 @@
+/* SPDX-License-Identifier: BSD-3-Clause
```
[the remark this answers is not in the document]
Comment (KeithWiles) [reply lib/cnet/incs/cnet_rarp.h:1]: This code was split out from the files, I did not add any new code, but will move this out into a new PR.

File: lib/core/pktmbuf/pktmbuf_ptype.c [hunk 15]
```diff
@@ -264,3 +275,470 @@ cne_get_ptype_name(uint32_t ptype, char *buf, size_t buflen)
 
     return 0;
 }
+
```
[the remark this answers is not in the document]
Comment (KeithWiles) [reply lib/core/pktmbuf/pktmbuf_ptype.c:278]: These were moved from the cne_net.h file this I believe, we seemed to have it split between pktmbuf and clib/net

File: lib/include/net/cne_ether.h (hunk 10 above)
Comment (KeithWiles) [reply lib/include/net/cne_ether.h:1]: This is not new code, but removed from other files and combined into this one file.

File: lib/include/net/cne_net.h (hunk 11 above)
Comment (KeithWiles) [reply lib/include/net/cne_net.h:44]: This code came from DPDK some time back, but will remove intel from the names.

File: usrtools/txgen/app/cli-functions.c (hunk 13 above)
Comment (KeithWiles) [reply usrtools/txgen/app/cli-functions.c:194]: Converted to a strtol() call instead as it returns an if found.

File: lib/cnet/arp/cnet_arp.c [hunk 16]
```diff
@@ -91,12 +92,15 @@ cnet_arp_add(int netif_idx, struct in_addr *addr, struct ether_addr *mac, int pe
...
+                     inet_ntop4(ipaddr, sizeof(ipaddr), &entry->pa, &mask));
 
         idx = ret;
         if (cne_fib_add(fi->fib, addr->s_addr, 32, idx)) {
             fib_info_free(fi, idx);
-            CNE_NULL_RET("ARP add failed for %s\n", inet_ntop4(&entry->pa, &mask));
+            cnet_arp_free(entry);
+            CNE_NULL_RET("ARP add failed for %s\n",
+                         inet_ntop4(ipaddr, sizeof(ipaddr), &entry->pa, &mask));
         }
```
Comment (maryamtahhan) [inline lib/cnet/arp/cnet_arp.c:104]: Referencing entry after free...

**Review by jeffreybshaw — APPROVED** [review]
